frontend/src/Components/Portal.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+interface PortalProps {
+  children: React.ReactNode;
+  target?: Element | DocumentFragment;
+}
+
+function Portal(props: PortalProps) {
+  const { children, target = document.getElementById('portal-root') } = props;
```

Comment:
You might want to move the `getElementById` call above the component to avoid calling it for every portal